src/ibex_bluesky_core/devices/reflectometry/__init__.py

```diff
@@ -0,0 +1,112 @@
+    StandardReadableFormat,
+    observe_value,
+)
+from ophyd_async.epics.core import epics_signal_r, epics_signal_w
+
+from ibex_bluesky_core.utils import get_pv_prefix
+
+logger = logging.getLogger(__name__)
+
+
+class ReflParameter(StandardReadable):
+    """Utility device for a reflectometry server parameter."""
+
+    def __init__(self, prefix: str, name: str, changing_timeout_s: float) -> None:
+        """Reflectometry server parameter.
+
+        Args:
+            prefix: the PV prefix.
+            name: the name of the parameter.
+            changing_timeout_s: seconds to wait for the CHANGING signal to go to False after a set.
+
+        """
+        with self.add_children_as_readables(StandardReadableFormat.HINTED_SIGNAL):
+            self.readback: SignalR[float] = epics_signal_r(float, f"{prefix}REFL_01:PARAM:{name}")
+        self.setpoint: SignalW[float] = epics_signal_w(float, f"{prefix}REFL_01:PARAM:{name}:SP")
+        self.changing: SignalR[bool] = epics_signal_r(
+            bool, f"{prefix}REFL_01:PARAM:{name}:CHANGING"
+        )
+        self.redefine = ReflParameterRedefine(prefix=f"{prefix}REFL_01:PARAM:{name}:", name="")
+        self.changing_timeout = changing_timeout_s
+        super().__init__(name=name)
+        self.readback.set_name(name)
+
+    @AsyncStatus.wrap
+    async def set(self, value: float) -> None:
+        """Set the setpoint.
+
+        This waits for the reflectometry parameter's 'CHANGING' PV to go True to
+         indicate it has finished.
+        """
+        logger.info("setting %s to %s", self.setpoint.source, value)
+        await self.setpoint.set(value, wait=True, timeout=None)
+        await asyncio.sleep(0.1)
+        logger.info("waiting for %s", self.changing.source)
+        async for chg in observe_value(self.changing, done_timeout=self.changing_timeout):
+            logger.debug("%s: %s", self.changing.source, chg)
+            if not chg:
+                break
+
+    def __repr__(self) -> str:
+        """Debug representation."""
+        return f"{self.__class__.__name__}(name={self.name})"
+
+
+class ReflParameterRedefine(StandardReadable):
+    """Utility device for redefining a reflectometry server parameter."""
+
+    def __init__(self, prefix: str, name: str, changed_timeout_s: float = 5.0) -> None:
+        """Reflectometry server parameter redefinition.
+
+        Args:
+            prefix: the reflectometry parameter full address.
+            name: the name of the parameter redefinition.
+            changed_timeout_s: seconds to wait for the CHANGED signal to go True after a set.
+
+        """
+        self.changed: SignalR[bool] = epics_signal_r(bool, f"{prefix}DEFINE_POS_CHANGED")
+        self.define_pos_sp = epics_signal_w(float, f"{prefix}DEFINE_POS:SP")
+        self.changed_timeout = changed_timeout_s
+        super().__init__(name)
+
+    @AsyncStatus.wrap
+    async def set(self, value: float) -> None:
+        """Set the setpoint.
```

Comment:
I still think this should be more explicit that it's _redefining_ the position of the axis when this `set()` method gets called - I know it's in a `ReflParameterRedefine` device but given redefining can lead to all sorts of subtle stuff, I'd rather be explicit